lib-injection/sitecustomize.py

```diff
@@ -33,55 +33,65 @@ def _log(msg, *args, level="info"):
     print("%s:datadog.autoinstrumentation(pid: %d): " % (level.upper(), os.getpid()) + msg % args, file=sys.stderr)
 
 
-try:
-    import ddtrace
-except ModuleNotFoundError:
-    _log("user-installed ddtrace not found, configuring application to use injection site-packages")
-
-    platform = "manylinux2014" if _get_clib() == "gnu" else "musllinux_1_1"
-    _log("detected platform %s" % platform, level="debug")
-
-    script_dir = os.path.dirname(__file__)
-    pkgs_path = os.path.join(script_dir, "ddtrace_pkgs")
-    _log("ddtrace_pkgs path is %r" % pkgs_path, level="debug")
-    _log("ddtrace_pkgs contents: %r" % os.listdir(pkgs_path), level="debug")
-
-    python_version = ".".join(str(i) for i in sys.version_info[:2])
-    site_pkgs_path = os.path.join(pkgs_path, "site-packages-ddtrace-py%s-%s" % (python_version, platform))
-    _log("site-packages path is %r" % site_pkgs_path, level="debug")
-    if not os.path.exists(site_pkgs_path):
-        _log("ddtrace site-packages not found in %r" % site_pkgs_path, level="error")
-
-    # Add the custom site-packages directory to the Python path to load the ddtrace package.
-    sys.path.insert(0, site_pkgs_path)
-    _log("sys.path %s" % sys.path, level="debug")
 
+def _inject():
     try:
-        import ddtrace  # noqa: F401
-
-    except BaseException as e:
-        _log("failed to load ddtrace module: %s" % e, level="error")
-        raise
+        import ddtrace
+    except ModuleNotFoundError:
+        _log("user-installed ddtrace not found, configuring application to use injection site-packages")
+
+        platform = "manylinux2014" if _get_clib() == "gnu" else "musllinux_1_1"
+        _log("detected platform %s" % platform, level="debug")
+
+        script_dir = os.path.dirname(__file__)
+        pkgs_path = os.path.join(script_dir, "ddtrace_pkgs")
+        _log("ddtrace_pkgs path is %r" % pkgs_path, level="debug")
+        _log("ddtrace_pkgs contents: %r" % os.listdir(pkgs_path), level="debug")
+
+        python_version = ".".join(str(i) for i in sys.version_info[:2])
+        if python_version not in ("3.7", "3.8", "3.9", "3.10", "3.11"):
+            _log("unsupported Python version %r, aborting" % python_version, level="error")
+            return
```

Comment:
This diff is gnarly. The above three lines are the main addition.

The logic was put into a function so that we can short-circuit the logic when an error occurs.